### DISCUSSION TITLE
Bottom Topography in E3SM-2-0 CMIP6 Oceanic Output Data

### DISCUSSION THREAD

**Discussion by shunzi-work** [E3SM-Project/E3SM#6546] [E3SM Data help] [answered]
Hi everyone,

I’ve been working with the E3SM-2-0 3-D oceanic output data (e.g., thetao, so, etc.) available on the CMIP6 website (from picontrol run), and I’ve noticed that the data appears to be missing bottom topography. In other CMIP6 model outputs, values below the ocean floor are typically marked as "NaN," which allows for an easy distinction between active cells and non-active regions (sea floor). However, in the E3SM-2-0 output, these values are set to 0. From what I understand, these datasets are regridded from the raw vertical grid. As a result, there’s a layer of interpolated data between the actual values and the ocean floor (where values are 0) as showed in the following plot. This setup makes it challenging to extract the real active data by simply filtering out the zeros.

![image](https://github.com/user-attachments/assets/ea837e0b-ffb0-485b-bf96-4dae43583c4a)

Is this an expected characteristic of the E3SM-2-0 model output? Also, is there a separate dataset or processing step that incorporates bottom topography? If this behavior is intentional, what’s the recommended approach for those who work exclusively with the data from the active cells?

I’m not sure if this is the correct place to raise this issue. If not, could someone please direct me to the appropriate channel?

Thank you!

**Comment by xylar** [comment, marked as answer]
@shunzi-work, it looks like I really messed up!  Before v2.0, we had a masking process in place that took care of bathymetry.  During some clean up in preparation for v2.0, this masking code was removed and I didn't do my due diligence as the quality control person for the ocean team at catching the consequences.

We're going to work to fix this output.

I'm about to head out on travel but I will try to put together a 3D fraction field tomorrow that should have been used to renormalize the data (and mask it where it is invalid.  I can share that with you if it would be helpful.  Otherwise, you would need to wait until the data can be reprocessed with the correct masking.

**Reply by shunzi-work** [reply]
Thank you for the quick response! The 3D fraction field sounds very helpful. I’d appreciate it if you could share it when it’s ready, and if there are any specific instructions on how to apply it effectively, that would be greatly appreciated as well. 
However, if it’s too much of an inconvenience, I’m also willing to wait for the corrected data. 
Thanks again for your help!

**Reply by xylar** [reply]
Hi @shunzi-work, I haven't had much luck so far producing a 3D fraction field.  I can produce one that I think is right, and I can show that it can renormalize a `thetao` field I produce myself.  But it doesn't produce sensible results for the `thetao` field that my colleagues have shared with me using the workflow used to produce the files you are looking at.  I will keep looking into it but it seems like my idea of retroactively renormalizing the data we have already published may not be as easy as I hoped.  I will keep you posted but currently it looks like you might have to wait for the data to be corrected on ESGF.

**Reply by xylar** [reply]
@shunzi-work, an update that we have fixed the masking issue in our post-processing software:
https://github.com/E3SM-Project/e3sm_to_cmip/pull/270
I don't think we have an estimate yet on how soon the files will be fixed.  There are a *lot* of them and they're pretty big so it'll take some time to chug through them.

@TonyB9000, could you let @shunzi-work know once the E3SM v2.0 data has been reprocessed and uploaded to ESGF?

**Reply by TonyB9000** [reply]
@xylar @shunzi-work At present there are 400 CMIP6 (ocean 3D) datasets we need to regenerate.  The current focus are the v2_1 sets (80), and are being regenerated now (~12 are completed - estimate 1 week - maybe 2 weeks - for all 80).  Then I will dive into the remaining 320 v2_0 sets.  If any particular experiments or variables are a priority, I can move them up to the head of the (v2) line, no problem (just a rearrangement of the ordering of the dataset_ids supplied to the generation processing).

**Reply by shunzi-work** [reply]
@TonyB9000 @xylar Thank you for the update. I'm happy to wait till all new datasets are regenerated. Will all new datasets be updated on ESGF at the same time or part by part?

**Reply by TonyB9000** [reply]
@shunzi-work @xylar I am in the process of regenerating the 80 v2_1 sets.  I will publish those as soon as they complete.  It will take longer to regnerate the v2 and v2-NARRM sets, and I can regernate (and publish) them in parts, if there is any priority desired.

**Reply by shunzi-work** [reply]
@TonyB9000 Thank you! I'd appreciate it if you can prioritize the outputs from "piControl" experiment from v2_0 and v2_0-NARRM.
